packages/manager/modules/pci/src/projects/project/private-networks/private-networks.service.js

```diff
@@ -24,18 +21,37 @@ export default class {
 
   getPrivateNetworks(serviceName) {
     return this.$http
-      .get(`/cloud/project/${serviceName}/network/private`)
-      .then(({ data }) =>
-        sortBy(data, 'vlanId').filter(
-          ({ status }) => !DELETING_STATUS.includes(status),
-        ),
-      );
+      .get(`/cloud/project/${serviceName}/aggregated/network`)
+      .then(({ data }) => {
+        const privateNetworks = {};
+        data.resources.forEach((network) => {
+          if (network.visibility === 'private') {
+            if (!privateNetworks[network.vlanId]) {
+              const { id, region, ...rest } = network;
+              privateNetworks[network.vlanId] = {
+                ...rest,
+                subnets: [{ region, networkId: id }],
+              };
+            } else {
+              const { id, region } = network;
+              privateNetworks[network.vlanId].subnets.push({
+                region,
+                networkId: id,
+              });
+            }
+          }
+        });
+        return sortBy(Object.values(privateNetworks), 'vlanId');
+      });
```

Comment:
It would be possible to do with .filter() and .map() rather than nesting if?

Reply:
If I use .filter() and then .map(), I will be iterating twice. So I have avoided it by achieving the desired result only in 1 iteration.